### PR TITLE
fix: send $reset_async_payment command for payment after authenticate

### PR DESCRIPTION
**Pull request by awwit**
Reset the payment status after authorizing the transaction. This is necessary to update the payment status after retrying payment.

Fixed a bug where payment status was retrieved from the wrong object.

Test added: `should reset async payment after authenticate`.

Eslint is added and configured.

Updated dependencies for building the module: `rollup` and plugins.

Updated `babel` and `webpack`.

### CODE REVIEW

File: src/payment/card/stripe.js [hunk 1]
```diff
@@ -119,7 +120,23 @@ export default class StripeCardPayment extends Payment {
...
 
-    return this._confirmCardPayment(intent);
+    try {
+      const result = await this._confirmCardPayment(intent);
+
+      return result;
+    } finally {
+      await this._resetAsyncPayment(payment.id);
+    }
+  }
```
Comment (awwit) [inline src/payment/card/stripe.js:130]: These are the main changes.

File: src/payment/card/stripe.js [hunk 2]
```diff
@@ -166,6 +186,6 @@ export default class StripeCardPayment extends Payment {
       throw new Error(actionResult.error.message);
     }
 
-    return { status: actionResult.status };
+    return { status: actionResult.paymentIntent.status };
```
Comment (awwit) [inline src/payment/card/stripe.js:189]: This is an additional tax.

File: src/payment/card/stripe.js [hunk 3]
```diff
@@ -119,7 +120,23 @@ export default class StripeCardPayment extends Payment {
 
     await this.loadScripts(this.scripts);
 
-    return this._confirmCardPayment(intent);
+    try {
+      const result = await this._confirmCardPayment(intent);
+
+      return result;
+    } finally {
+      await this._resetAsyncPayment(payment.id);
```
Comment (swellmike) [inline src/payment/card/stripe.js:128]: Should we only reset payment status on error?

File: src/payment/card/stripe.js [hunk 4]
```diff
@@ -119,7 +120,23 @@ export default class StripeCardPayment extends Payment {
...
+
+  /**
+   * Reset the payment timer to update the payment status faster
+   *
+   * @param {string} id
+   * @returns {Promise<object>}
+   */
+  _resetAsyncPayment(id) {
+    return this.request('put', '/payments', id, { $async_reset: true });
   }
```
Comment (vstudenichnik-insoft) [inline src/payment/card/stripe.js:140]: I think it's better to move this method as public to the payments module to call it for any payment method (`swell.payment.resetAsyncPayment(id)`).

File: src/payment/card/stripe.js (hunk 3 above)
Comment (awwit) [reply src/payment/card/stripe.js:128]: No, if we want to get the transaction state as quickly as possible, then we should reset the asynchronous payment state.

For example, if a payment has been pending for a long time, then we will need to wait some time to receive an updated transaction status.

If we reset the asynchronous payment state, we will get a new transaction state after 30 seconds.

**Review by swellmike — APPROVED** [review]
approved pending operator name change

**Pull request merged**